### DISCUSSION TITLE
How can I config multithreading?

### DISCUSSION THREAD

**Discussion by ghost** [shadowsocks/shadowsocks-rust#1134] [Q&A] [answered]
Some doubts about the parameters for setting the number of threads.
```
--worker-threads <WORKER_THREADS>
          Sets the number of worker threads the `Runtime` will use
```

1, How many threads **ss-server** uses by default?
2, If my single-core processor is relatively idle, how much do you recommend setting the number of threads, to 4, 8, 16, or 32?
3, Is it a good practice to open multiple **ssserver** processes to listen to multiple ports on one single-core processor machine?

**Comment by database64128** [comment, marked as answer]
This is why I'm convinced that this option shouldn't have existed at all. We have seen more than a few times, that people with zero understanding of coroutines or scheduling are getting confused by this option and tend to misconfigure it.

To answer your question:
1. With the `trust-dns` feature enabled, the number of tokio's **worker threads** equals the number of CPU threads on your system.
2. Just pretend that this option does not exist and do not think about it at all.
3. It'd be more efficient to use the `servers` array in config to define multiple servers for a single `ssserver` instance to run, because userspace coroutine scheduling is much more efficient than kernel thread scheduling.

**Reply by ghost** [reply]
Thanks for your answer, it's very helpful
